Content.Shared/Pinpointer/PinpointerComponent.cs

```diff
@@ -12,25 +12,46 @@ namespace Content.Shared.Pinpointer
     public sealed class PinpointerComponent : Component
     {
         // TODO: Type serializer oh god
-        [DataField("component")]
+        [DataField("component"), ViewVariables(VVAccess.ReadWrite)]
         public string? Component;
 
-        [DataField("mediumDistance")]
+        [DataField("mediumDistance"), ViewVariables(VVAccess.ReadWrite)]
         public float MediumDistance = 16f;
 
-        [DataField("closeDistance")]
+        [DataField("closeDistance"), ViewVariables(VVAccess.ReadWrite)]
         public float CloseDistance = 8f;
 
-        [DataField("reachedDistance")]
+        [DataField("reachedDistance"), ViewVariables(VVAccess.ReadWrite)]
         public float ReachedDistance = 1f;
 
         /// <summary>
         ///     Pinpointer arrow precision in radians.
         /// </summary>
-        [DataField("precision")]
+        [DataField("precision"), ViewVariables(VVAccess.ReadWrite)]
         public double Precision = 0.09;
 
+        /// <summary>
+        ///     When true, ignores <see cref="Component"/> and allows <see cref="Target"/> to be
+        ///     set by using the entity on the desired target.
+        /// </summary>
+        [DataField("emgagged"), ViewVariables(VVAccess.ReadWrite)]
+        public bool Emagged;
```

Comment:
This should just be a separate EmaggedComponent. I'm not gonna make you update old comps for it.

Reply:
generic like #13867 (or if someone makes better one then that)